include/kos/thread.h

```diff
@@ -269,18 +295,19 @@ typedef struct kthread_attr {
     const char *label;
 } kthread_attr_t;
 
-/** \name   Threading system modes
-    \brief  kthread mode values
+/** \brief  kthread mode values
+
+    \deprecated
+    Only preemptive scheduling is still supported!
 
     The threading system will always be in one of the following modes. This
     represents either pre-emptive scheduling or an un-initialized state.
-
-    @{
 */
-#define THD_MODE_NONE       -1  /**< \brief Threads not running */
-#define THD_MODE_COOP       0   /**< \brief Cooperative mode \deprecated */
-#define THD_MODE_PREEMPT    1   /**< \brief Preemptive threading mode */
-/** @} */
+typedef enum kthread_mode {
```

Comment:
Because I don't want to make this entire file pretty, idiomatic C and then just randomly leave this one deprecated piece using a completely different coding style... Seemed very inconsistent, and at least if you're using a deprecated API, you're doing it with more strongly typed enumerations.

Also enumerations are far more useful in terms of documentation and doxygen searching and grouping... Not that anyone should be searching for this since it's deprecated... lol.